CONTRIBUTING.md

```diff
@@ -3,8 +3,26 @@
 Thank you for your interest in contributing to our project. Whether it's a bug report, new feature, correction, or additional
 documentation, we greatly value feedback and contributions from our community.
 
-Please read through this document before submitting any issues or pull requests to ensure we have all the necessary
-information to effectively respond to your bug report or contribution.
+Please read through this document before submitting any issues or pull requests to ensure we have all the necessary information to effectively respond to your bug report or contribution.
+
+## Getting Started
+
+If you're looking to join this wonderful wave of collaboration, here's how to begin:
+
+1. Take a look at the open issues to see what needs attention.
+
+2. **Important: Comment Before You Start Working**
+   - comment on an issue before you start working on it
+   - Wait for a maintainer to officially assign the issue to you
+   - This prevents duplicate work and ensures you have the latest context
+
+3. **If an Issue is Already Assigned**
+   - If you see an issue you're interested in but it's assigned to someone else
+   - Check the last activity date
+   - If there haven't been recent updates, you can politely comment asking both the assignee and maintainers about the status
+   - Example: "Hi! I notice this issue hasn't had recent activity. @assignee, are you still working on this? If not, would it be okay if I took it over?"
+
+4. Once assigned, you'll receive the green light to begin your contribution
```

Comment:
Once assigned, is there a subsequent step? Or, is the assignment the green light?